barretenberg/cpp/pil/vm2/opcodes/get_env_var.pil

```diff
@@ -0,0 +1,152 @@
+ * |   3   | chainId        |         0         |          1          |          0         | CHAIN_ID_ROW_IDX  |     0      |     0     |        0         |       0       |         0       |      0       |      0         |
+ * |   4   | version        |         0         |          1          |          0         | VERSION_ROW_IDX   |     0      |     0     |        0         |       0       |         0       |      0       |      0         |
+ * |   5   | blockNumber    |         0         |          1          |          0         | BLOCK_NUM_ROW_IDX |     0      |     0     |        0         |       0       |         0       |      0       |      0         |
+ * |   6   | timestamp      |         0         |          1          |          0         | TIMESTAMP_ROW_IDX |     0      |     0     |        0         |       0       |         0       |      0       |      0         |
+ * |   7   | feePerL2Gas    |         0         |          0          |          1         | GAS_FEES_ROW_IDX  |     0      |     0     |        0         |       1       |         0       |      0       |      0         |
+ * |   8   | feePerDaGas    |         0         |          1          |          0         | GAS_FEES_ROW_IDX  |     0      |     0     |        0         |       0       |         0       |      0       |      0         |
+ * |   9   | isStaticCall   |         0         |          0          |          0         |         0         |     0      |     0     |        0         |       0       |         1       |      0       |      0         |
+ * |  10   | l2GasLeft      |         0         |          0          |          0         |         0         |     0      |     0     |        0         |       0       |         0       |      1       |      0         |
+ * |  11   | daGasLeft      |         0         |          0          |          0         |         0         |     0      |     0     |        0         |       0       |         0       |      0       |      1         |
+ * |  12+  | (invalid)      |         1         |          0          |          0         |         0         |     0      |     0     |        0         |       0       |         0       |      0       |      0         |
+ * +-------+----------------+-------------------+---------------------+--------------------+-------------------+------------+-----------+------------------+---------------+-----------------+--------------+----------------+
+ */
+
+namespace execution; // this is a virtual gadget that shares rows with the execution trace
+
+    // No relations will be checked if this identity is satisfied.
+    #[skippable_if]
+    sel_get_env_var = 0; // from execution.pil.
+
+    // Selector for when we should get environment variable (gated by earlier errors)
+    pol commit sel_should_get_env_var;
+    sel_should_get_env_var * (1 - sel_should_get_env_var) = 0;
```

Comment:
Do you need this boolean constraint? You seem to fully constrain it in `#[SHOULD_GET_ENV_VAR]`